### PR TITLE
[3231] Add update service to allocations

### PR DESCRIPTION
**Pull request by SamJamCul**
### Context

https://trello.com/c/mC7AqJvX/3231-s-11-change-my-allocations-request

### Changes proposed in this pull request

* Add update service to set number of places appropriately for allocation

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

**Comment by SamJamCul** [comment]
[[S] 11. Change my allocations request](https://trello.com/c/mC7AqJvX/3231-s-11-change-my-allocations-request)

File: app/services/allocations/update.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  private
+
+    def set_number_of_places
+      object.number_of_places = case params[:request_type]
+                                when "declined"
+                                  0
+                                when "repeat"
+                                  object.previous&.number_of_places || 0
+                                else
+                                  params[:number_of_places]
```
Comment (defong) [inline app/services/allocations/update.rb:28]: so if its not

`declined` or `repeat` what is is it?

https://github.com/DFE-Digital/teacher-training-api/blob/3dbee3b7ef813cd7006927e9c7a75315d2b0d787/spec/factories/allocations.rb#L7-L12

Feels head scratching it is missing a trait as well as a test for the other `request_type`

File: app/services/allocations/update.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      @params = params
+
+      set_number_of_places
+      set_request_type
+
+      object
+    end
+
+    def execute
+      object.save
```
Comment (defong) [inline app/services/allocations/update.rb:16]: why isn't this return value tested?

File: app/services/allocations/update.rb (hunk 1 above)
Comment (SamJamCul) [reply app/services/allocations/update.rb:28]: The third option for request_type is `initial` but we're going to be addressing this in another PR. Updating an initial request to set a different number of places is going to require some further logic here, and we aren't quite sure yet what it'll look like.

File: app/services/allocations/update.rb (hunk 2 above)
Comment (SamJamCul) [reply app/services/allocations/update.rb:16]: it it now, thanks

**Pull request merged**